### DISCUSSION TITLE
Deno KV: consider using LMDB instead of SQLite for local

### DISCUSSION THREAD

**Discussion by alexgleason** [denoland/deno#18975] [Ideas]
LMDB is embedded like sqlite, is already a key-value store instead of a relational database, and is much faster in benchmarks: https://github.com/kriszyp/lmdb-js

**Comment by alexgleason** [comment]
The author of lmdb-js thinks this is a good idea. The databases are nearly 1:1 according to him: https://github.com/kriszyp/lmdb-js/discussions/236#discussioncomment-6016979

sqlite seems like kind of a hack for local dev. Backing it with lmdb would make self-hosted Deno KV suitable for production workloads.

**Reply by bartlomieju** [reply]
CC @losfair

**Comment by xyzshantaram** [comment]
Hi all, I've started working on this in a personal fork of Deno (it's very barebones for now). I'm building a new KV backend implementation that will slot in alongside the existing SQLite and remote implementations. 

@bartlomieju @losfair Would you guys be interested in merging something like that? I'd like to know before I continue working on it because forking would (obviously) not be feasible for me and I'd rather just do it as an FFI lib if it's not going to end up inside Deno.

https://github.com/xyzshantaram/denokv_lmdb is where it's at, it's very early stages at the moment.

**Reply by bartlomieju** [reply]
@xyzshantaram could you please open an issue about it to discuss in more detail?

**Reply by xyzshantaram** [reply]
Opened #23545, please have a look at your convenience. Thank you!
